Neos.Fusion/Classes/Core/Parser.php

```diff
@@ -46,6 +46,34 @@ class Parser implements ParserInterface
      */
     protected $parserCache;
 
+    /**
+     * Parses the given Fusion source code, resolves includes and returns a merged array tree
+     * as the result.
+     *
+     * @param FusionSourceCode|FusionSourceCodeCollection $sourceCode The Fusion source code to parse
+     * @return array The merged array tree for the Fusion runtime, generated from the source code
+     * @throws Fusion\Exception
+     * @api
+     */
+    public function parseFrom(FusionSourceCode|FusionSourceCodeCollection $sourceCode): array
```

Comment:
you _could_ introduce a common interface for the two that allows iterating (for both, just one entry for the single thing).

Reply:
yes why not ;)

Reply:
I find it a challeging idea ... I don't think there can be a common interface ...
I just created for fun an "Eierlegendewollmilchsau" dto:

```php
class Dto implements IteratorAggregate
{
  private function __construct(private string $name, private int $value, private array $dtos) {}

  public static function create(string $name, int $value): Dto {
    return new self($name, $value);
  }

  public static function fromDtos(Dto ...$dtos): Dto {
    return new self(dtos: $dtos);
  }

  public function getName(): string
  {
     return $this->name ?? $this->dtos[0]->name;
  }

  public function getValue(): string
  {
     return $this->value = $this->dtos[0]->value;
  }

  public function getIterator(): ArrayIterator {
    return new ArrayIterator($this->dtos ?? [$this]);
  }
}
```

this seems like the worst idea one can have :D as you'd never know if you hold a real iterable or the final values ^^

Reply:
Yeah obviously not a single object, but if you implement an recursive iterator across both and make that a common interface, so that you can just take whatever object you have at hand and do `foreach ($fusionCollection as $name => $value)` or something and it gives you just recursively all `FusionSourceCode` objects in that construct, that should work out fine.

Reply:
I just thought aswell that a FusionSourceCodeInterface that is implemented by `FusionSourceCode` and the `FusionSourceCodeCollection` might be a good idea. However this would have to include the method "getContextPath" which makes no sense on collection level at all and this is what this pr is about. 

So i get that the collection is just a collection.

Reply:
Doesn't need to be one interface for everything, for this specific method at least just the iteration is important, and as said, I could see a way to create an interface that allows you to iterate and only returns FusionCode objects so that at that point you can be sure to have getContextPath